features/notifications/notifications_controller.rb

```diff
@@ -0,0 +1,83 @@
+require_relative "../../shared/authenticated_controller_base"
+require_relative "../../services/services"
+require_relative "../../taskqueue/task_queue"
+require "pathname"
+
+module FastlaneCI
+  # CRUD controller for handling and displaying notifications to user
+  class NotificationsController < AuthenticatedControllerBase
+    HOME = "/notifications"
+
+    # Instantiates a new `Notification Controller` with a task queue to process
+    # requests
+    #
+    # @param  [Sinatra::Base] app
+    def initialize(app = nil)
+      super(app)
+      @task_queue = TaskQueue::TaskQueue.new(name: "notifications")
+    end
+
+    # Renders the notifications dashboard, displaying a table of notifications
+    # scoped to the user
+    get HOME do
+      notifications = Services.notification_service.notification_data_source.notifications
+      user_notifications = notifications.select { |notification| notification.user_id == user.id }
+      locals = { notifications: user_notifications, title: "Notifications" }
+      erb(:dashboard, locals: locals, layout: FastlaneCI.default_layout)
+    end
+
+    # Creates a new notification
+    post "#{HOME}/create" do
+      add_to_task_queue do
+        payload = notification_params(request)
+        Services.notification_service.create_notification!(payload)
```

Comment:
Yeah, like here, I think we actually want ` Services.notification_service.create_notification!` to handle the async dispatching